### PR TITLE
Add missing mandatory file extensions for ESM JavaScript

### PR DESCRIPTION
**Pull request by vanitabarrett**
Fixes https://github.com/alphagov/govuk-frontend/issues/2633 

> A file extension must be provided when using the import keyword to resolve relative or absolute specifiers. Directory indexes (e.g. './startup/index.js') must also be fully specified.
https://nodejs.org/api/esm.html#mandatory-file-extensions

This PR has been updated to go with Option 2, where our JavaScript files are `.mjs` by default.

--- 

Two approaches here at the moment:

- Option 1: minimal changes to the pre-built files, just adding the missing `.js` file extensions. This means we need to update these file extensions when we copy the files in `package/govuk-esm/` so that they reference `.mjs` rather than `.js`
- Option 2: pre-built files use `.mjs` as they're already in ESM format. These can be copied directly over to `package/govuk-esm/` with no changes. UMD build (using rollup) gets rid of (resolves) import statements anyway and transform to `.js`.

### CODE REVIEW

**Comment by 36degrees** [comment]
Agreed that option 2 seems like the more 'honest' approach, and also puts us on the right track to eventually move entirely to ESM.

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,19 @@ We've deprecated the `govuk-header__link--service-name` class, and will remove i
...
+#### File extensions added for JavaScript ES Module imports
+In 4.1.0, we started providing our component JavaScript in ES module format. This JavaScript used import statements without [mandatory file extensions](https://nodejs.org/api/esm.html#mandatory-file-extensions), as required by Node.js.
+
+We have now updated our component JavaScript to include these missing file extensions. If you previously received an error similar to the below, this fix should resolve that issue:
+
+```
+Cannot find module '../node_modules/govuk-frontend/govuk-esm/common' imported from ../node_modules/govuk-frontend/govuk-esm/all.mjs
+```
+
+If you are already successfully importing our JavaScript as ES modules, you should not need to make any changes, but there may be config you can remove. For example, removing `fullySpecified: false` from your Webpack config file.
```
Comment (domoscargin) [inline CHANGELOG.md:24]: My only thought is that this is under "Recommended changes", and we get right up to this line before we suggest an actual change.

Are we able to expand on an entry in "Fixes"? Or is that not really something we do? We could possibly include this action here, and the rest of the info under fixes.

**Comment by domoscargin** [comment]
Small update needed in the Gulp Tasks doc (.js to .mjs)

https://github.com/alphagov/govuk-frontend/blob/da32bb0e9700da1aeb020f83a94d28f3393bfbcb/docs/contributing/tasks.md?plain=1#L62

**Review by domoscargin — APPROVED** [review]
I've left a very minor comment, but otherwise this all looks good to me, and still works as expected locally. 🎉

**Comment by vanitabarrett** [comment]
Great spot @domoscargin , thanks! Will change that now

**Pull request merged**